tests/utils/object_store/test_object_store.py

```diff
@@ -55,11 +88,13 @@ def object_store(request, monkeypatch: pytest.MonkeyPatch,
         os.makedirs(remote_dir)
         object_store_kwargs[request.param]['provider_kwargs']['key'] = remote_dir
         yield request.param(**object_store_kwargs[request.param])
+    elif request.param is SFTPObjectStore:
+        yield MockSFTPObjectStore('test_hostname', port=24, username='test_user')
```

Comment:
Instead of creating the `MockSFTObjectStore`, can we use `monkepatch` and use the original object store class?

Something like:

```
elif request.param is SFTPObjectStore:
    # setup the server
    server = mockssh.Server(users={})
    private_key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
    pem = private_key.private_bytes(encoding=serialization.Encoding.PEM,
                                    format=serialization.PrivateFormat.TraditionalOpenSSL,
                                    encryption_algorithm=serialization.NoEncryption())
    with tempfile.TemporaryDirectory() as tmpdir:
            tmppath = os.path.join(tmpdir, 'test_rsa_key')
            with open(tmppath, 'wb') as private_key_file:
                private_key_file.write(pem)
            server.add_user(uid=self.username, private_key_path=tmppath)
            with server:
                client = server.client(username)
                # Trick the SFTPObjectStore to create a client attached to the mock server
                monekpatch.setattr(composer.utils.object_store.sftp_object_store, "SSHClient", lambda: client)
                 yield SFTPObjectStore(**object_store_kwargs[request.param])

Reply:
It might be necessary to refactor this logic out into a shared helper, so it can also be used for the test_object_store_hparams_is_constructable test below.

Reply:
I'm still going to monkeypatch the whole function because I also don't want the key_file_authentication commands to run. (They still would if I only patch the client). However, I will get rid of the class.